### ISSUE TITLE
Incosistent provider score

### ISSUE THREAD

**Issue by lobis** [w3f/1k-validators-be#2816]
Sometimes the provider score is not what it should be, I think because the providers are not being registered correctly. (I am using kusama.w3f.community as the source of truth, if this is not the case please point me towards a more accurate source for scoring).

For example, I have two validators under the 1KV programme and they both share location and provider and they are both in the 1.9.0 version (latest at the time of writing).

- https://kusama.w3f.community/candidate/E1iq5eqQqTz8dM1P1bqDEd5TtS2DK2cHLapcsMqh6CnRJkV. Here the provider score is  98.94736842105263 which I believe to be correct. More importantly, in this case the provider information is also shown.
- https://kusama.w3f.community/candidate/GvG8TntkK5Lsg7DCA86MKYNGyYnxktNk4cS1x115eiQfRpT. In this case the provider score is 25 points (remember both validators use the same provider). In this case the provider information is missing.

In summary one of my validators is getting less points than it should due to missing provider information. I have reported this in the past via email but I decided to open an issue to better keep track of progress.

Thanks!

**Comment by lobis** [comment]
Also see https://github.com/w3f/1k-validators-be/issues/2807.

**Comment by ironoa** [comment]
Hey lobis,
just double checked and I noticed that:
- the two validators have a similar provider score
- the two validators have a different provider
<img width="411" alt="image" src="https://github.com/w3f/1k-validators-be/assets/25957738/967f1c5b-490e-43c4-b61d-4ca7237a4adf">
<img width="411" alt="image" src="https://github.com/w3f/1k-validators-be/assets/25957738/1709f7a7-a653-4c33-95a8-47caaeec5b6e">

**Comment by lobis** [comment]
> Hey lobis, just double checked and I noticed that:
> 
> * the two validators have a similar provider score
> * the two validators have a different provider
> 
> <img alt="image" width="411" src="https://private-user-images.githubusercontent.com/25957738/320436678-967f1c5b-490e-43c4-b61d-4ca7237a4adf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NzA2NjIsIm5iZiI6MTcxMjU3MDM2MiwicGF0aCI6Ii8yNTk1NzczOC8zMjA0MzY2NzgtOTY3ZjFjNWItNDkwZS00M2M0LWI2MWQtNGNhNzIzN2E0YWRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDA5NTkyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwYTAyZjY0ZTQxMTQ0MDkzOTRkZjYwNjM3YmU2MWZmN2I4ZmJjOGEzNWZlNjZhY2E1NGQ0YjBkY2E5MDk2YzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I_Q1X7z8pBwzFaLY2k2qRkc8CUakkBg2_1VTcXvKbDs"> <img alt="image" width="411" src="https://private-user-images.githubusercontent.com/25957738/320436760-1709f7a7-a653-4c33-95a8-47caaeec5b6e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NzA2NjIsIm5iZiI6MTcxMjU3MDM2MiwicGF0aCI6Ii8yNTk1NzczOC8zMjA0MzY3NjAtMTcwOWY3YTctYTY1My00YzMzLTk1YTgtNDdjYWFlZWM1YjZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDA5NTkyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNDMyZGRlYjAxYjUxMDU5OTAwMzlmOTg0ZTU4MjBjZWVjYzc5MjhhZmNmZDFlMTEyYzBhM2E4OTYwMDQ0MjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WJ1WgXuKQ48XXPvhq1fQsW08fKhjekrnO4Kc_xT6yZA">

Yes, since I wasn't getting the correct provider score I deployed the validator in the aws cloud for a few hours, then redeployed to the original location. The provider score updated to the aws location but never updated back to what it should've been in the first place (the same one as my other validator). Atleast now I am getting the points I should be getting so I can't complain about that.

However this second validator (the one with the provider issue) seems to still be having problems with the scoring backend, since it's been a long time since it gets any nomination and it's now averaging around 800 points. It was recently nominated but lost the nomination after 2 eras, all while maintaining a very high score. For instance my other validator has lower score and is consistently nominated for more time, so i would love to know what is going on (I guess this last part would be more relevant in https://github.com/w3f/1k-validators-be/issues/2807).

**Comment by lobis** [comment]
> Hey lobis, just double checked and I noticed that:
> 
> * the two validators have a similar provider score
> * the two validators have a different provider

It appears it's broken again, I don't have a provider anymore.

**Comment by lobis** [comment]
After trying different things out, my conclusion is that the issue only arises when multiple validators share the same ip. In my case I was self-hosting both validators in my home network. I assume this it not intended so it's a bug.

@ironoa I'll close the issue if you say this is intended behaviour. I'll look into migrating one of my validators into the cloud so I can get proper scoring.

**Comment by ironoa** [comment]
you are right @lobis thanks for investigating, [this PR](https://github.com/w3f/1k-validators-be/pull/2853) addresses the `multiple validators share the same ip` bug. It will be soon deployed (currently under testing on staging)

**Comment by lobis** [comment]
Thanks! Everything seems fine from my end.
